internal/aws/shield.go

```diff
@@ -0,0 +1,53 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/shield"
+)
+
+type ShieldAPI interface {
+	ShieldAvailable(ctx context.Context) (bool, error)
+	GetSubscriptionStatus(ctx context.Context) (*shield.GetSubscriptionStateOutput, error)
+	GetProtection(ctx context.Context, resourceArn *string) (*shield.Protection, error)
+	CreateProtection(ctx context.Context, resourceArn *string) (*shield.CreateProtectionOutput, error)
+	DeleteProtection(ctx context.Context, protectionID *string) (*shield.DeleteProtectionOutput, error)
+}
+
+func (c *Cloud) ShieldAvailable(ctx context.Context) (bool, error) {
+	status, err := c.GetSubscriptionStatus(ctx)
+	if err != nil {
+		return false, err
```

Comment:
We don't check error type here. In happy case it seems to be not found, but what if its something else?

Reply:
The only service specific GetSubscriptionState error case seems to be InternalErrorException. Added extra test cases.